File: Hotel/Entities/Transaction.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Hotel.Entities
{
    public class Transaction
    {
        public int TransactionId { get; set; }
        public DateTime CheckInTime { get; set; }
        public DateTime CheckOutTime { get; set; }

        public virtual User User { get; set; }
        [JsonIgnore] public virtual Room Room { get; set; }

        public int UserId { get; set; }
        public int RoomId { get; set; }
        public int Cost { get; set; }
        public bool IsPaid { get; set; }
        public bool IsCanceled { get; set; }

    }
}

File: Hotel/ClientApp/src/app/transaction/models/transaction.ts
import { jsonIgnoreReplacer, jsonIgnore } from 'json-ignore';

export class Transaction {

  public transactionId: number;
  public checkInTime: Date;
  public checkOutTime: Date;
  public userId: number;
  public roomId: number;
  public cost: number;
  public isPaid: Boolean;
  public isCanceled: Boolean;
  public isEvicted: Boolean;

  @jsonIgnore() public TheNoumber: number;
  @jsonIgnore() public Loading: Boolean;
}

export interface TransactionFilter {

  clientId: number;
  checkInTime: Date;
  checkOutTime: Date;
  type: String;

}

export class FinancicalInformation {

  public dateTime: Date;
  public sum: Number;

  @jsonIgnore() public TheNoumber: number;
}

export interface FinancicalInfoFilter {

  start: Date;
  end: Date;

}

export interface ExpandData {

  transactionId: Number;

  userName: String;
  birthDate: Date;
  email: String;
  phone: String;
  clientID: String;

  roomName: String;
  floor: String;
  roomTypeId: Number;
  numberOfSeats: Number;
  hasMiniBar: Boolean;

}

